### PR TITLE
Cleanup Order. 

### PR DESCRIPTION
**Pull request by doctrino**
## Description

To refactor the `load_data_models` I need an explicit way of handling dependencies, so introduced this mechanism. 

Will take the refactoring of `load_data_models` in a separate PR to keep the size small

Pure refactoring so no changelog entry. 

## Checklist:
- [x] Tests added/updated.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped. [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: tests/test_approval_modules_snapshots/example_pump_asset_hierarchy.yaml [hunk 1]
```diff
@@ -3,45 +3,6 @@ DataSet:
   externalId: src:lift_pump_stations
   name: Lift Pump Stations
 Row:
-- args: []
```
Comment (doctrino) [inline tests/test_approval_modules_snapshots/example_pump_asset_hierarchy.yaml:6]: Same here, a bug in causing `RAW` tables to be written twice.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ repos:
           - --fix
           - --exit-non-zero-on-fix
           - --line-length=120
-          - --ignore=E731,E501,W605,T201
+          - --ignore=E731,E501,W605,T201,UP007
```
Comment (doctrino) [inline .pre-commit-config.yaml:10]: Needed as ruff wants `Optional[str]` to be written `str | None`, which is not supported by `typer`, were we use it for example 
```python
include: Annotated[
        Optional[list[str]],	        Optional[list[str]],
        typer.Option(	        typer.Option(
            "--include",	            "--include",
            "-i",	            "-i",
            help=f"Specify which resources to deploy, available options: {_AVAILABLE_DATA_TYPES}",	            help=f"Specify which resources to deploy, available options: {_AVAILABLE_DATA_TYPES}",
        ),	        ),
    ] = None,	    ] = None,
```

File: cognite_toolkit/cdf.py [hunk 3]
```diff
@@ -256,85 +255,42 @@ def deploy(
...
-        alternatives = {
-            folder.name: f"{folder.parent.name}/{folder.name}"
-            for folder in Path(build_dir).parent.iterdir()
-            if folder.is_dir()
-        }
-        matches = difflib.get_close_matches(Path(build_dir).name, list(alternatives.keys()), n=3, cutoff=0.3)
-        print(
-            f"  [bold red]WARNING:[/] {build_dir} does not exists. Did you mean one of these? {[alternatives[m] for m in matches]}"
-        )
-        exit(1)
```
Comment (doctrino) [inline cognite_toolkit/cdf.py:281]: I removed this as I think it is no longer helpful. The build directory should be `build` 99% of the time.

File: cognite_toolkit/cdf_tk/load.py [hunk 4]
```diff
@@ -227,168 +228,13 @@ def delete(self, ids: Sequence[T_ID]) -> int:
     def retrieve(self, ids: Sequence[T_ID]) -> T_ResourceList:
         return self.api_class.retrieve(ids)
 
-    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> T_Resource | T_ResourceList:
+    def load_resource(self, filepath: Path, ToolGlobals: CDFToolConfig) -> T_Resource | T_ResourceList:
```
Comment (doctrino) [inline cognite_toolkit/cdf_tk/load.py:231]: Renaming to make this more descriptive.

File: cognite_toolkit/cdf_tk/load.py [hunk 5]
```diff
@@ -227,168 +228,13 @@ def delete(self, ids: Sequence[T_ID]) -> int:
...
         raw_yaml = load_yaml_inject_variables(filepath, ToolGlobals.environment_variables())
         if isinstance(raw_yaml, list):
             return self.list_cls.load(raw_yaml)
         return self.resource_cls.load(raw_yaml)
 
 
-@final
-class TimeSeriesLoader(Loader[str, TimeSeries, TimeSeriesList]):
```
Comment (doctrino) [inline cognite_toolkit/cdf_tk/load.py:238]: Reordering is necessary as `TimeSeriesLoader` is dependent on `DataSetLoader`, so `DataSetLoader` must come before `TimeSeriesLoader` in the file.

**Review by ronpal — APPROVED** [review]
Very nice!

**Pull request merged**